Automattic/wp-calypso
# Staging Sites: Handle lock expiration

Related to https://github.com/Automattic/dotcom-forge/issues/5801

## Proposed Changes

This PR handles the lock expiration in the UI.

## Testing Instructions
1. Create a site and make it atomic.
3. Apply diff: D139827-code
4. Return early in `wp-content/mu-plugins/site-staging/class.site-staging.php` after line 155 `self::add_staging_site_creation_lock( $blog_id );` to trigger the error
5. Press the Create a staging site button.
6. Ensure that `/lock` endpoint should get polled, and it'll end up sending a nonnull response
7. After 20 seconds ensure that the response of lock is `null` and the UI shows again the Create staging site button.



## Pre-merge Checklist
